packages/theme-check-common/src/checks/valid-schema-name/index.spec.ts

```diff
@@ -16,10 +16,70 @@ describe('Module: ValidSchemaName', () => {
 
     const offenses = await runLiquidCheck(ValidSchemaName, sourceCode, DEFAULT_FILE_NAME);
     expect(offenses).toHaveLength(1);
-    expect(offenses[0].message).toEqual('Schema name is too long (max 25 characters)');
+    expect(offenses[0].message).toEqual(
+      "Schema name 'test-schema-name-that-is-too-long' is too long (max 25 characters)",
+    );
 
     const highlights = highlightedOffenses({ [DEFAULT_FILE_NAME]: sourceCode }, offenses);
     expect(highlights).to.have.length(1);
     expect(highlights[0]).toBe('"test-schema-name-that-is-too-long"');
   });
+
+  it('reports no offenses with schema name translation that exists and is not over 25 chars long', async () => {
+    const offenses = await check(
+      {
+        'locales/en.default.schema.json': '{ "my_translation_key": "My translation is good."}',
+        'code.liquid': `
+          {% schema %}
+            {
+              "name": "t:my_translation_key"
+        } 
+      {% endschema %}`,
+      },
+      [ValidSchemaName],
+    );
+
+    expect(offenses).toHaveLength(0);
+  });
+
+  it('reports an offense with schema name translation is missing', async () => {
+    const offenses = await check(
+      {
+        'locales/en.default.schema.json': '{ "another_translation_key": "Another translation"}',
+        'code.liquid': `
+          {% schema %}
+            {
+              "name": "t:my_translation_key"
+            } 
+          {% endschema %}`,
+      },
+      [ValidSchemaName],
+    );
+
+    expect(offenses).toHaveLength(1);
+    expect(offenses[0].message).toEqual(
+      "'t:my_translation_key' does not have a matching entry in 'locales/en.default.schema.json'",
+    );
+  });
+
+  it('reports an offense with schema name translation that exists and is over 25 chars long', async () => {
+    const offenses = await check(
+      {
+        'locales/en.default.schema.json':
+          '{ "my_translation_key": "My translation is tooooooooo long."}',
```

Comment:
Can you add a nested property test? `getTranslation` looks fine, but probably better to at least have 1 test.

Reply:
I also think we probably have a similar util somewhere. Something called `deepGet` or something.